tests/regression/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+    builder.build()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crabgrind as cg;
+    use s2n_tls::testing::TestPair;
+    use std::env;
+    use std::process::Command;
+
+    const COST: u64 = 1_000_000; //configurable threshold for regression
+    // environment variable to determine whether to run under valgrind or solely test functionality
+    fn is_running_under_valgrind() -> bool {
+        env::var("VALGRIND").is_ok()
+    }
+    //test to create new config, set security policy, host_callback information, load/trust certs, and build config
+    #[test]
+    fn test_set_config() -> Result<(), s2n_tls::error::Error> {
+        if !is_running_under_valgrind() {
+            cg::cachegrind::stop_instrumentation();
+            cg::cachegrind::start_instrumentation();
+            let keypair_rsa = CertKeyPair::default();
+            let _config = set_config(&security::DEFAULT_TLS13, keypair_rsa).expect("Failed to build config");
+            return Ok(());
+        }
+        run_valgrind_test("test_set_config");
+        Ok(())
+    }
+
+    #[test]
+    fn test_rsa_handshake() -> Result<(), s2n_tls::error::Error> {
+        if !is_running_under_valgrind() {
+            cg::cachegrind::stop_instrumentation();
+            // Example usage with RSA keypair (default)
+            let keypair_rsa = CertKeyPair::default();
+            let config = set_config(&security::DEFAULT_TLS13, keypair_rsa)?;
+            // Create a pair (client + server) using that config, start handshake measurement
+            let mut pair = TestPair::from_config(&config);
+            // Assert a successful handshake
+            cg::cachegrind::start_instrumentation();
+            assert!(pair.handshake().is_ok());
+            cg::cachegrind::stop_instrumentation();
+            return Ok(());
+        }
+        run_valgrind_test("test_rsa_handshake");
+        Ok(())
+    }
+    // function to run specified test using valgrind
+    fn run_valgrind_test(test_name: &str) {
+        let exe_path = std::env::args().next().unwrap();
+        let output_file = format!("cachegrind_{}.out", test_name);
+        let valgrind_command = format!(
+            "valgrind --tool=cachegrind --cachegrind-out-file={} {} {}",
+            output_file, exe_path, test_name
+        );
+    
+        println!("Running command: {}", valgrind_command);
+        let output_command = format!("--cachegrind-out-file={}", &output_file);
+        let status = Command::new("valgrind")
+            .args(&["--tool=cachegrind", &output_command, &exe_path, test_name])
+            .env_remove("VALGRIND") //ensures that the recursive call is made to the actual harness code block rather than back to this function
+            .status()
+            .expect("Failed to execute cargo test");
+    
+        if !status.success() {
+            panic!("Valgrind failed");
+        }
+    
+        let annotate_file = format!("perf_outputs/{}.annotated.txt", test_name);
```

Comment:
this should be written to `target`. That's the convention of where to put derived artifacts for rust projects.